tests/apps/core/test_core_api_groups.py

```diff
@@ -0,0 +1,307 @@
+        self.assertEqual(group.name, "The group")
+
+    def test_api_groups_update_administrator_of_another(self):
+        """Being administrator of a group should not grant
+        authorization to update another group."""
+        group = GroupFactory(name="Old name")
+        admin = UserFactory()
+        jwt_token = AccessToken.for_user(admin)
+
+        response = self.client.put(
+            f"/api/groups/{group.id}/",
+            {"name": "The group"},
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response.json(),
+            {"detail": "You do not have permission to perform this action."},
+        )
+        group.refresh_from_db()
+        self.assertEqual(group.name, "Old name")
+
+    def test_api_groups_delete_authenticated(self):
+        """Authenticated users should not be allowed to delete
+        a room they are not an administrator of."""
+        group = GroupFactory()
+        user = UserFactory()
+        jwt_token = AccessToken.for_user(user)
+
+        response = self.client.delete(
+            f"/api/groups/{group.id}/",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response.json(),
+            {"detail": "You do not have permission to perform this action."},
+        )
+        self.assertEqual(Group.objects.count(), 1)
+
+    def test_api_groups_delete_administrator(self):
+        """Administrators should be allowed to delete a group."""
+        admin = UserFactory()
+        group = GroupFactory(members=[(admin, True)])
+        jwt_token = AccessToken.for_user(admin)
+
+        response = self.client.delete(
+            f"/api/groups/{group.id}/",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(Group.objects.count(), 0)
+
+    def test_api_groups_delete_administrator_of_another(self):
+        """Being administrator of a group should not grant
+        authorization to delete another group."""
+        group = GroupFactory()
+        admin = UserFactory()
+        jwt_token = AccessToken.for_user(admin)
+
+        response = self.client.delete(
+            f"/api/groups/{group.id}/",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response.json(),
+            {"detail": "You do not have permission to perform this action."},
+        )
+        self.assertEqual(Group.objects.count(), 1)
+
+    def test_api_groups_add_members_authenticated(self):
+        """Authenticated users should not be allowed to add
+        members to a group they are not an administrator of."""
+        group = GroupFactory()
+        users = [UserFactory(), UserFactory(), UserFactory()]
```

Comment:
Nothing important, but you can do:
```suggestion
        users = UserFactory.create_batch(3)
```